docs/site/migration/express-middleware.md

```diff
@@ -6,8 +6,242 @@ sidebar: lb4_sidebar
+    this.app = express();
+    this.lbApp = new CoffeeShopApplication(options);
+
+    // Middleware migrated from LoopBack 3
+    this.app.use(loopback.favicon());
+    this.app.use(compression());
+    this.app.use(cors());
+    this.app.use(helmet());
+
+    // Mount the LB4 REST API
+    this.app.use('/api', this.lbApp.requestHandler);
+
+    // Custom Express routes
+    this.app.get('/ping', function(_req: Request, res: Response) {
+      res.send('pong');
+    });
+
+    // Serve static files in the public folder
+    this.app.use(express.static(path.join(__dirname, '../public')));
+  }
+
+  public async boot() {
+    await this.lbApp.boot();
+  }
+
+  public async start() {
+    await this.lbApp.start();
+    const port = this.lbApp.restServer.config.port || 3000;
+    const host = this.lbApp.restServer.config.host || '127.0.0.1';
+    this.server = this.app.listen(port, host);
+    await pEvent(this.server, 'listening');
+  }
+
+  public async stop() {
+    if (!this.server) return;
+    await this.lbApp.stop();
+    this.server.close();
+    await pEvent(this.server, 'close');
+    this.server = undefined;
+  }
+}
+```
+
+This app will replace the `CoffeeShopApplication` as the entry point for the
+program, modify the `src/index.ts` file accordingly.
+
+{% include code-caption.html content="src/index.ts" %}
+
+```ts
+import {ApplicationConfig} from '@loopback/core';
+import {ExpressServer} from './server';
+
+export {ExpressServer};
+
+export async function main(options: ApplicationConfig = {}) {
+  const server = new ExpressServer(options);
+  await server.boot();
+  await server.start();
+  console.log('Server is running at http://127.0.0.1:3000');
+}
+```
+
+Next, modify the `index.js` file in the root of the project to prevent the LB4
+app from listening, by adding `listenOnStart: false` in `config.rest` object.
+The `config` object should now look like this:
+
+{% include code-caption.html content="index.js" %}
+
+```js
+const config = {
+  rest: {
+    port: +process.env.PORT || 3000,
+    host: process.env.HOST || 'localhost',
+    openApiSpec: {
+      // useful when used with OpenAPI-to-GraphQL to locate your application
+      setServersFromRequest: true,
+    },
+    listenOnStart: false,
+  },
+};
+```
+
+Then, in the `bootOptions` of the `CoffeeShopApplication` class, add the
+`lb3app` to configure the path of the LB3 APIs.
+
+{% include code-caption.html content="src/application.ts" %}
+
+```js
+lb3app: {
+  mode: 'fullApp';
+}
+```
+
+`this.bootOptions` should now look like this:
+
+{% include code-caption.html content="src/application.ts" %}
+
+```ts
+this.bootOptions = {
+  controllers: {
+    // Customize ControllerBooter Conventions here
+    dirs: ['controllers'],
+    extensions: ['.controller.js'],
+    nested: true,
+  },
+  lb3app: {
+    mode: 'fullApp',
+  },
+};
+```
+
+Start the app:
+
+```sh
+$ npm start
+```
+
+Load http://localhost:3000/ on your browser. This will load the Express app,
```

Comment:
Please replace the naked URL with a markdown link.

```suggestion
Load [http://localhost:3000/](http://localhost:3000/) on your browser. This will load the Express app,
```